src/server/transaction.cc

```diff
@@ -190,32 +190,22 @@ void Transaction::InitGlobal() {
 }
 
 void Transaction::BuildShardIndex(const KeyIndex& key_index, std::vector<PerShardCache>* out) {
+  // Because of the way we iterate in InitShardData
+  DCHECK(!key_index.bonus || key_index.step == 1);
+
   auto& shard_index = *out;
+  for (unsigned i : key_index.Range()) {
+    string_view key = ArgS(full_args_, i);
+    unique_slot_checker_.Add(key);
+    ShardId sid = Shard(key, shard_data_.size());
 
-  auto add = [&shard_index](uint32_t sid, uint32_t b, uint32_t e) {
+    unsigned step = shard_index[sid].key_step = key_index.bonus ? 1 : key_index.step;
```

Comment:
is it `=` or `==` here?

Reply:
it's =, I just re-use the variable step below, while also storing in at shard_index[sid].key_step

Reply:
please separate this.